apps/sc-app/src/test/MyLeg.tsx

```diff
@@ -9,13 +15,41 @@ interface OwnProps {
 
 export function MyLeg(props: OwnProps): ReactElement {
     const { bet, handleDelete } = props;
+    const { state, getFreebet } = useFreebets();
+    const [freebet, setFreebet] = useState<Freebet | null>(null);
+
+    useEffect(() => {
+        const selectedFreebet = getFreebet(`${bet.id}`);
+
+        if (isUndefined(selectedFreebet)) {
+            setFreebet(null);
+            return;
+        }
+
+        setFreebet(selectedFreebet);
+    }, [bet.id, state.timestamp]);
 
     return (
         <div style={{ textAlign: 'left', display: 'flex' }} data-testid={`bet-${bet.id}`}>
-            <div>
-                bet: {bet.event.name}
-                <input type="text" name={`bet-${bet.id}-stake`} />
-            </div>
+            <FreeBetContainer linkedEntityId={`${bet.id}`} />
```

Comment:
notice the prop name - it's part of the story - to be able to trace the connection between new feature and the existing app/code